src/sources/kubernetes/test.rs

```diff
@@ -277,6 +299,19 @@ impl Drop for Kube {
     }
 }
 
+/// If F returns None, retries it after some time, for some count,
+/// returning None if all trys fail.
+fn retry<F: FnMut() -> Option<R>, R>(mut f: F) -> Option<R> {
```

Comment:
I wonder if it would make sense to take advantage of https://github.com/timberio/vector/blob/master/src/test_util.rs#L163 and wait for it to exist. We could probably modify this to get a return value.